src/main/java/soconnect/storage/TodoListStorage.java

```diff
@@ -18,18 +19,19 @@ public interface TodoListStorage {
     Path getTodoListFilePath();
 
     /**
-     * Returns TodoList data as a {@link ReadOnlyTodoList}.
+     * Returns {@code TodoList} data as a {@link ReadOnlyTodoList}.
      * Returns {@code Optional.empty()} if storage file is not found.
      *
      * @throws DataConversionException If the data in storage is not in the expected format.
      * @throws IOException If there was any problem when reading from the storage.
      */
-    Optional<ReadOnlyTodoList> readTodoList() throws DataConversionException, IOException;
+    Optional<ReadOnlyTodoList> readTodoList(ReadOnlySoConnect soConnect) throws DataConversionException, IOException;
 
     /**
-     * @see #readTodoList()
+     * @see #readTodoList(ReadOnlySoConnect)
      */
-    Optional<ReadOnlyTodoList> readTodoList(Path filePath) throws DataConversionException, IOException;
+    Optional<ReadOnlyTodoList> readTodoList(ReadOnlySoConnect soConnect, Path filePath) throws DataConversionException,
+            IOException;
```

Comment:
Is the parameter given correct? If I remember correctly `ReadOnlySoConnect` is for the contact list, there is another `ReadOnlyTodoList` or something right?

Reply:
Originally, this method is supposed to take in no arguments. But when the `Tag`s got implemented, there arose a need to verify that the `Tag` data in `todolist.json` are actual `Tag`s. (i.e. They need to match the tags in the tags list in `soconnect.json`). Thus, the `ReadOnlySoConnect` here is passed along until it reaches `JsonSerializableTodoList.toModelType` where the verification occurs.